R/yearmonth.R

```diff
@@ -65,8 +65,27 @@ yearmonth.Date <- function(x) {
 
 #' @export
 yearmonth.character <- function(x) {
-  assertDate(x)
-  new_yearmonth(anydate(x))
+  key_words <- regmatches(x, gregexpr("[[:alpha:]]+", x))
+  if (all(grepl("^[[:digit:]]{1~4}[[:space:]]*(m|mon|month)[[:space:]]*[[:digit:]]{1~4}$",
```

Comment:
Can you assign the regular expression to a variable?
Is it too strict to include `[[:space:]]` here?

Reply:
The reason to include `[[:space]]*` is to allow the possibility, since I've seen both `2018M01` and `2018 M01`. I haven't seen `2018M 1` before, but it's easy enough to allow for, and I can imagine `2018 month 1` in a spreadsheet (probably created by concatenation).

Reply:
The initial reason I decided to go with such a strict format compared to the quarterly thing, honestly, is that `m` is a far more common letter than `q`. I'm far more worried about something that isn't a date but looks like one for the first few lines with `m` than `q`.

Reply:
I've assigned the regular expression to a variable in a subsequent commit.

Reply:
Including `[[:space:]]*` doesn't seem to make any difference between `2018M01` and `2018M01`. Can you remove `[[:space:]]*`?

Reply:
Done and done.